src/components/Table/Table.tsx

```diff
@@ -1,74 +1,131 @@
 import { Table as MantineTable, TextInput } from "@mantine/core";
-import { ReactNode, useState } from "react";
+import { type ReactNode, useMemo, useState } from "react";
+import { IconSortAscending, IconSortDescending } from "@tabler/icons-react";
 
 interface Column {
   columnName: string;
   exactMatch: boolean;
   header: string;
-  cellRenderer?: (row: any, column: Column) => ReactNode;
+  cellRenderer?: (row: any) => ReactNode;
 }
 interface TableProps<T extends { id: number }> {
   data: T[];
   columns: Column[];
 }
 
+interface Filter {
+  name: "asc" | "desc" | null;
+}
+interface Sort {
+  name: "asc" | "desc" | null;
+}
+
 export function Table<T extends { id: number }>(props: TableProps<T>) {
-  const [data, setData] = useState(props.data);
-  const headers = props.columns.map((column, index) => (
-    <th key={index}>{column.header}</th>
-  ));
+  const [filtersState, setFilters] = useState<Record<string, Filter>>({});
+  const [sortState, setSorting] = useState<Record<string, Sort>>({});
 
-  const handleFilterChange = (event) => {
-    const filterValue = event.target.value;
-    const currentColumn = event.target.name.toLowerCase();
-    if (filterValue !== "") {
-      const columns = props.columns;
-      const filteredData = props.data.filter((value) => {
-        for (let i = 0; i < columns.length; i++) {
-          if (currentColumn === columns[i].columnName.toLowerCase()) {
-            if (columns[i].exactMatch)
-              return value[columns[i].columnName].toString() === filterValue;
-            else
-              return value[columns[i].columnName]
-                .toString()
-                .includes(filterValue);
-          }
-        }
-      });
-      setData(filteredData);
-    } else setData(props.data);
+  const handleSort = (event, columnName: string) => {
+    setSorting((prevSorting) => {
+      if (sortState[columnName] === undefined) sortState[columnName] = "asc";
```

Comment:
`sortState[columnName] = "asc"` is a no-no, since you're updating the reference of the current state.

Any mutations should be done on new obj references:

```
setSorting((prevSortSate) => {
 const updatedSortState = { ...prevSortSate }; // <-- creating new obj reference
 
 if (updatedSortState[columnName] === undefined) updatedSortState[columnName] = ...

 ...
})
```